homeassistant/components/cover/ryobi_gdo.py

```diff
@@ -0,0 +1,285 @@
+"""
+Ryobi platform for the cover component.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/ryobiGDO/
+"""
+import logging
+import json
+import socket
+from datetime import timedelta
+import voluptuous as vol
+import requests
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.cover import (
+    CoverDevice, PLATFORM_SCHEMA, SUPPORT_OPEN, SUPPORT_CLOSE)
+from homeassistant.helpers.event import track_utc_time_change
+from homeassistant.const import (
+    CONF_DEVICE, CONF_USERNAME, CONF_PASSWORD,
+    STATE_UNKNOWN, STATE_CLOSED, STATE_OPEN, CONF_COVERS)
+
+REQUIREMENTS = ['websocket-client==0.37.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+COVER_SCHEMA = vol.Schema({
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+})
```

Comment:
The current configuration looks like that every door needs its own username/password combination. I assume that you can have multiple doors with one account. Would something like this make more sense than adding the credentials for every unit?

```yaml
cover:
  - platform: ryobi_gdo
    username: USERNAME
    password: PASSWORD
    device_id:
      - DEVICE_ID_1
      - DEVICE_ID_2
```

Or think about adding all doors by default for a first integration.

Reply:
your are right, you can have multiple doors with one account. I choose that to be able to use multiple ryobi accounts in one configuration. with for example on device per username. Using your proposed configuration I guess that I can't use multiple accounts ?